### PR TITLE
Remove code that can be replaced with URL API.

### PR DESCRIPTION
**Pull request by davidlehn**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +36,12 @@ Forge ChangeLog
   Depending on how applications used this id to name association it could cause
   compatibility issues.
 
+### Notes
+- The URL related changes may expose bugs in some of the networking related
+  code (unrelated to the much wider used cryptography code). The automated and
+  manual test coverage for this code is weak at best. Issues or patches to
+  update the code or tests would be appriciated.
```
Comment (dlongley) [inline CHANGELOG.md:43]:
```suggestion
  update the code or tests would be appreciated.
```

File: lib/log.js [hunk 2]
```diff
@@ -298,15 +298,20 @@ if(typeof(console) !== 'undefined' && 'log' in console) {
  * that could otherwise be limited by a user config.
  */
 if(sConsoleLogger !== null) {
-  var query = forge.util.getQueryVariables();
-  if('console.level' in query) {
+  var query;
+  if(typeof(window) !== 'undefined' && window.location) {
```
Comment (dlongley) [inline lib/log.js:302]:
```suggestion
  if(typeof window !== 'undefined' && window.location) {
```

**Review by dlongley — APPROVED** [review]
+1 for using WHATWG URL and requiring a polyfill if it isn't present on the target system.

**Comment by ready-research** [comment]
@davidlehn Can you please confirm this issue in [huntr ](https://www.huntr.dev/bounties/41852c50-3c6d-4703-8c55-4db27164a4ae/). Thanks,

**Comment by davidlehn** [comment]
@ready-research Done. No idea what huntr is doing here, but it dropped the fix bounty, oh well. I doubt anyone was actually calling that code anyway.  Best it was removed.

**Pull request merged**